### PR TITLE
Vault v2 storage updates & refactor

### PR DESCRIPTION
**Pull request by aalavandhan**
* Added fees and keeper role
* Added capital control hyper-parameters
* using `uint8` for iterators
* Internal function refactor, broke up recover function into multiple methods

### CODE REVIEW

File: spot-contracts/contracts/vaults/RolloverVault.sol [hunk 1]
```diff
@@ -125,6 +131,50 @@ contract RolloverVault is
...
+    /// @return The address of the keeper.
+    address public keeper;
+
+    /// @notice The enforced minimum balance of underlying tokens to be held by the vault at all times.
+    /// @dev On deployment only the delta greater than this balance is deployed.
+    uint256 public minUnderlyingBal;
+
+    /// @notice The enforced maximum percentage of the vault's TVL that can be held as underlying tokens.
+    /// @dev When the users meld or swap assets with the vault, the vault effectively exchanges tranches
+    ///      for more liquidity that can be used for future deployments. This parameter acts controls
```
Comment (nms-7) [inline spot-contracts/contracts/vaults/RolloverVault.sol:147]:
```suggestion
    ///      for more liquidity that can be used for future deployments. This parameter controls
```

**Review by nms-7 — APPROVED** [review]

**Review by brandoniles — COMMENTED** [review]
Looks good, just minor comments

File: spot-contracts/contracts/vaults/RolloverVault.sol [hunk 2]
```diff
@@ -125,6 +131,50 @@ contract RolloverVault is
...
+    ///      the extent to which the vault can allow melding or swapping.
+    uint256 public maxUnderlyingPerc;
+
+    struct FeeData {
```
[the remark this answers is not in the document]
Comment (aalavandhan) [reply spot-contracts/contracts/vaults/RolloverVault.sol:151]: Not necessarily, we can break it up into individual variables. The struct merely looks cleaner ..

File: spot-contracts/contracts/vaults/RolloverVault.sol [hunk 3]
```diff
@@ -125,6 +131,50 @@ contract RolloverVault is
...
+        // @notice The percentage of vault notes withheld as fees on redemption.
+        uint256 redemptionFeePerc;
+        // @notice The maximum percentage fee paid in the underlying assets by users who meld.
+        // @dev The final fee is discounted based on remaining to maturity.
+        uint256 meldFeePerc;
+        // @notice The maximum percentage fee paid in the underlying assets by users who swap.
+        // @dev The final fee is discounted based on remaining to maturity.
+        uint256 swapFeePerc;
+        // @notice A fixed cost charged at the time of deployment denominated in the underlying asset.
+        uint256 protocolFee;
```
[the remark this answers is not in the document]
Comment (aalavandhan) [reply spot-contracts/contracts/vaults/RolloverVault.sol:161]: Agree

File: spot-contracts/contracts/vaults/RolloverVault.sol [hunk 4]
```diff
@@ -79,9 +90,9 @@ contract RolloverVault is
 
     //-------------------------------------------------------------------------
     // Constants
-    uint8 public constant PERC_DECIMALS = 6;
-    uint256 public constant UNIT_PERC = 10**PERC_DECIMALS;
-    uint256 public constant HUNDRED_PERC = 100 * UNIT_PERC;
+    uint8 public constant PERC_DECIMALS = 8;
+    uint256 public constant UNIT_PERC = 10**(PERC_DECIMALS - 2);
+    uint256 public constant HUNDRED_PERC = 10**PERC_DECIMALS;
```
[the remark this answers is not in the document]
Comment (aalavandhan) [reply spot-contracts/contracts/vaults/RolloverVault.sol:95]: I've gone back and forth on this. Agree `ONE_PERCENT` is clearer, we can go back and update all instances in a separate PR ..

File: spot-contracts/contracts/vaults/RolloverVault.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (aalavandhan) [reply spot-contracts/contracts/vaults/RolloverVault.sol:152]: Alternatively we could move it out to the an external fee strategy too

**Pull request closed without merging**